lib/customerio/client.rb

```diff
@@ -182,8 +209,15 @@ def extract_attributes(args)
       hash.inject({}){ |hash, (k,v)| hash[k.to_sym] = v; hash }
     end
 
+    def filter_recipients_data_for_field(recipients, field)
+      BROADCASTS_ALLOWED_RECIPIENT_FIELDS[field].reduce({}) do |obj, f|
```

Comment:
What are you trying to do here?

Reply:
This is the second part of formatting the `recipients` object for sending it to the API. It picks the acceptable fields from `BROADCASTS_ALLOWED_RECIPIENT_FIELDS` and builds a hash from the `recipients` fields.

```ruby
// input
{ emails: [ "niko@mlh.io"], foo: "bar", email_ignore_missing: true }

// output
{ emails: [ "niko@mlh.io"], email_ignore_missing: true }
```